sklearn/utils/extmath.py

```diff
@@ -499,6 +500,9 @@ def randomized_svd(
     >>> U.shape, s.shape, Vh.shape
     ((3, 2), (2,), (2, 4))
     """
+    xp, is_array_api_compliant = get_namespace(M)
+    M = check_array(M, accept_sparse=True)
```

Comment:
This new check will add a performance regression when `randomized_svd` is called via another public API (e.g. `PCA(solver="randomized").fit(X)`) that has already checked that the input is valid.

Could you instead please introduce a new private `_randomized_svd` with the actual implementation (and with a short docstring that refers to the the docstring of its public counterpart) and a public `randomized_svd` that does the check and then call into the private `_randomized_svd`.

Then callers of `randomized_svd` within the scikit-learn code base should be updated to instead call `_randomized_svd`.

Reply:
Since it's only called once in like PCA, I told myself that's fine and let it as is. But you're right, probably better to introduce private functions instead.